jest/jest.config.shared.js

```diff
@@ -23,9 +26,4 @@ module.exports = {
     "\\.[jt]sx?$": require.resolve("./transform"),
   },
   watchPathIgnorePatterns: [...ignorePatterns, "\\/node_modules\\/"],
-  watchPlugins: [
-    require.resolve("jest-watch-select-projects"),
-    require.resolve("jest-watch-typeahead/filename"),
-    require.resolve("jest-watch-typeahead/testname"),
-  ],
```

Comment:
I have no idea what these were doing but took them out to get rid of these warnings on running jest

![Screenshot 2024-04-01 at 2 56 52 PM](https://github.com/remix-run/react-router/assets/1609022/fe2f5ef9-0863-498e-bf7b-8de690d4fef2)

Feels smarter to decouple from jest-specific stuff too in anticipation of potentially moving to `vitest` or node's built-in test runner in the future